internal/access/access_key.go

```diff
@@ -56,12 +56,3 @@ func DeleteAccessKey(c *gin.Context, id uid.ID) error {
 
 	return data.DeleteAccessKeys(db, data.ByID(id))
 }
-
-func DeleteRequestAccessKey(c *gin.Context) error {
-	// does not need authorization check, this action is limited to the calling key
-	key := currentAccessKey(c)
-
-	db := getDB(c)
-
-	return data.DeleteAccessKey(db, key.ID)
-}
```

Comment:
I don't agree with this change. Once we allow `data` package calls in the handlers, it's really hard to tell if we're skipping `access` because there's no authorization requirements, or because the dev forgot.

Reply:
Let's continue to talk about this point. I think that because so many of the functions in `access` do not do access control, that signal of "this has correct authz because it calls access" is not very reliable. The changes in #2716 and #2717 show there are quite a few access functions that don't do any authorization.  Removing those functions makes the code a lot easier to trace, which should it easier to see how and where authz is implemented.

I think the guideline of "the first thing in the function is authorization" is a great idea, but hiding it in the access layer makes it less effective. If we bring it up into the API handlers it'll be much more obvious, and that will allow us to include required roles for an endpoint in the OpenAPI doc as well.

Reply:
Yeah, I don't agree with your assessment here. It's a lot easier to see something's missing in an empty function than missing at the start of a handler whose job is business logic and request translation; adding a third responsibility there muddies it, and devs are going to miss looking for it in review. It's not being hidden in the access layer, it's being explicit by putting it into its own layer. Empty functions are by design, and intentional, as they communicate intent effectively.

Don't get too attached to the current role-based design as it's not staying around long-term.